cypress/support/commands.ts

```diff
@@ -11,353 +11,277 @@ Cypress.Commands.add("clearIndexedDb", () => {
     });
 });
 
-Cypress.Commands.add("signup", () => {
-    cy.clearCookies();
-    cy.clearLocalStorage();
-    cy.clearIndexedDb();
+Cypress.Commands.add("clearEmails", () => {
+    return cy.request("DELETE", "http://localhost:1080/email/all");
+});
 
-    cy.visit("/");
+Cypress.Commands.add("getCodeFromEmail", (options: any = {}) => {
+    const getCode = () => {
+        return cy.request("http://localhost:1080/email").then((res) => {
+            const latest = res.body.sort((a, b) => (a.time > b.time ? -1 : 1))[0];
+            if (!latest) {
+                return null;
+            }
+            const matchCode = latest.text.match(/(\d{6})/);
+            return matchCode && matchCode[1];
+        });
+    };
+
+    const resolveValue = () => {
+        getCode().then((value) => {
+            // @ts-ignore
+            return cy.verifyUpcomingAssertions(value, options, {
+                onRetry: resolveValue,
+            });
+        });
+    };
```

Comment:
Had to use some Cypress-foo to add retry-ability here, since I didn't want to wait an arbitrary amount of time for the email to arrive. Instead, we're just retrying the maildev endpoint until we get a match (or the timeout elapses). Unfortunately adding retry-ability to custom commands is not well documented so I had to piece this together from built-in commands and answers on Github. To be honest I don't fully understand what's happening here, but it works 😊.

Reply:
Nice one!